### ISSUE TITLE
Hanging logging  screen while log page loading is finished in the background

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#7003]
##### Detailed steps causing the problem:
- Start logging a cache
- The logging screen is shown
- The progress spinner indicates that the log page is loaded from server in the background

##### Actual behavior after performing these steps:
As soon as loading of log page is finished, I experience an unresponsive behaviour for a second. 
I guess, that is because the screen is redrawn now showing the "Problem" dop down menu.

##### Expected behavior after performing these steps:
This is not a serious bug, however it is blocking the text input while typing, wich is kind of annoying and not a good user experience.

While not having an idea about what happens in background...isn't it somehow possible to show the "Problem" menu already at start of the logging activity (just silently assuming it will be supported) and only remove it in case it is not needed/not supported? That should make this need for redraw much less often.

##### Version of c:geo used:
2018.05.01-NB

##### Is the problem reproducible for you?
Yes

**Comment by Lineflyer** [comment]
As I have just seen, there is another drawback of the current working mode:
If you select TB actions, GCVoting and/or assign a favorite point before the loading has finished, those selections seem to be resetted once loading is finished.
That is similar to another issue (cant find it at the moment) where those fields are resetted when coming back to the logging screen after attaching a photo.

**Comment by pstorch** [comment]
I don't think this is introduced by the new logging page. I think it was a problem before, too. But I'm not sure.

**Comment by Lineflyer** [comment]
The loading in background was also performed before, but I cannot remember any impact to the user once loading is complete.

**Comment by Lineflyer** [comment]
At that time I had approx. 10 trackables in my inventory (both GK and TB). 
As discussed lets wait on the users feedback (now the version is live).

**Comment by Lineflyer** [comment]
Looking at Google Play Console this issue seems to get worse on high server load / unresponsive server as we have a high occurrence of ANRs reported on the weekends currently:

@pstorch 
Can you judge, whether we can do anything about it?

```
"main" tid=1 Waiting
"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x72a25710 self=0x74baac3a00
  | sysTid=7388 nice=-10 cgrp=default sched=0/0 handle=0x74bf27d9b0
  | state=S schedstat=( 30497540069 18862273863 71662 ) utm=2215 stm=834 core=0 HZ=100
  | stack=0x7ff6ce7000-0x7ff6ce9000 stackSize=8MB
  | held mutexes=
 
  at java.lang.Object.wait (Native method)
- waiting on <0x079f3bdc> (a java.lang.Object)
 
  at java.lang.Thread.parkFor$ (Thread.java:2135)
- locked <0x079f3bdc> (a java.lang.Object)
 
  at sun.misc.Unsafe.park (Unsafe.java:358)
 
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
 
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt (AbstractQueuedSynchronizer.java:868)
 
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1021)
 
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1328)
 
  at java.util.concurrent.CountDownLatch.await (CountDownLatch.java:232)
 
  at io.reactivex.internal.observers.BlockingMultiObserver.blockingGet (BlockingMultiObserver.java:83)
 
  at io.reactivex.Single.blockingGet (Single.java:2656)
 
  at cgeo.geocaching.connector.gc.GCWebAPI.getTrackableInventory (GCWebAPI.java:422)
 
  at cgeo.geocaching.connector.gc.GCLoggingManager.onLoadFinished (GCLoggingManager.java:76)
 
  at cgeo.geocaching.connector.gc.GCLoggingManager.onLoadFinished (GCLoggingManager.java:39)
 
  at android.support.v4.app.LoaderManagerImpl$LoaderInfo.callOnLoadFinished (LoaderManager.java:479)
 
  at android.support.v4.app.LoaderManagerImpl$LoaderInfo.onLoadComplete (LoaderManager.java:447)
 
  at android.support.v4.content.Loader.deliverResult (Loader.java:126)
 
  at android.support.v4.content.AsyncTaskLoader.dispatchOnLoadComplete (AsyncTaskLoader.java:249)
 
  at android.support.v4.content.AsyncTaskLoader$LoadTask.onPostExecute (AsyncTaskLoader.java:77)
 
  at android.support.v4.content.ModernAsyncTask.finish (ModernAsyncTask.java:476)
 
  at android.support.v4.content.ModernAsyncTask$InternalHandler.handleMessage (ModernAsyncTask.java:493)
 
  at android.os.Handler.dispatchMessage (Handler.java:105)
 
  at android.os.Looper.loop (Looper.java:251)
 
  at android.app.ActivityThread.main (ActivityThread.java:6572)
 
  at java.lang.reflect.Method.invoke (Native method)
 
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:767)
```

**Comment by pstorch** [comment]
I'm not sure, but it looks like the Logpage is loaded with an AsyncTask (`UrlLoader`) in the background, but when this call returns to the UI thread (`onLoadFinished()`) another web request is done to load the trackables (`GCWebAPI.getTrackableInventory()`). And if the second call takes a long time, the UI can freeze (ANR). The second call should be refactored and also use the Loader mechanism.

**Comment by Lineflyer** [comment]
This is no longer happening to me. Feel free to reopen if the problem is still present (maybe on less performant devices)
